### DISCUSSION TITLE
Representing resource limits with unhommogeneous task

### DISCUSSION THREAD

**Discussion by jossM** [apache/airflow#29527] [Q&A]
Hello!

We use redshift to process data and airflow to pilot it in our stack. Redshift has a configured limit to the number of query it runs in parrallel. To give more power to one particular query, we sometimes use several query slots using the wlm_slot_count paramater.
The amount of computing required for all kpis is important and our budget limited. Hence, we need to order the queries we run to respect our SLA.

From airflow standpoint we represented these query slots as pools and the use of multiple query slots as pools_slots. The ordering is currently handled via priority_weight. All in all the solution works well for all tasks except for the ones using large pools_slots. For these the orchestration logic faces [an issue where low prio tasks get computed before high prio & high pools_slots tasks](https://github.com/apache/airflow/issues/29474). As detailed in the case, it seems like this is not a undesired behaviour, but a design choice of airflow, hence it seems like we are representing our problem wrong in airflow.

I can think of a few hacks that we could do but that seems overly complicated, ex:
- We could put all tasks with a pools_slots of 1 and add one task directly upstream and directly downstream that would reduce the pool size by the expected pool slots-1
- We could have something that monitors high prio tasks waiting and kills running low prio tasks when a new tasks comes in (as suggested in the issue I believe)

What is the expected way to represent these contraints in airflow ?

Thanks for your opinion! 🙏

**Comment by potiuk** [comment]
This topic was further discussed in https://github.com/apache/airflow/issues/29474
